basenode/src/main/java/io/coti/basenode/services/BaseNodeFeesService.java

```diff
@@ -1,126 +1,140 @@
 
-import io.coti.basenode.crypto.CryptoHelper;
-import io.coti.basenode.data.FeeData;
-import io.coti.basenode.data.Hash;
-import io.coti.basenode.data.NodeFeeData;
-import io.coti.basenode.data.NodeFeeType;
-import io.coti.basenode.http.GetNodeFeesDataResponse;
-import io.coti.basenode.http.NodeFeeSetRequest;
-import io.coti.basenode.http.Response;
+import io.coti.basenode.crypto.OriginatorCurrencyCrypto;
+import io.coti.basenode.data.*;
+import io.coti.basenode.exceptions.CotiRunTimeException;
+import io.coti.basenode.http.*;
 import io.coti.basenode.http.interfaces.IResponse;
 import io.coti.basenode.services.interfaces.INodeFeesService;
+import io.coti.basenode.services.interfaces.ITokenFeeData;
 import lombok.extern.slf4j.Slf4j;
-import org.springframework.beans.factory.annotation.Value;
 import org.springframework.http.HttpStatus;
 import org.springframework.http.ResponseEntity;
 import org.springframework.stereotype.Service;
 
 import javax.validation.Valid;
 import java.math.BigDecimal;
-import java.math.RoundingMode;
-import java.nio.charset.StandardCharsets;
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.List;
 
+import static io.coti.basenode.crypto.CryptoHelper.calculateTokenFeeHash;
 import static io.coti.basenode.http.BaseNodeHttpStringConstants.INVALID_PARAMETERS_MESSAGE;
 import static io.coti.basenode.http.BaseNodeHttpStringConstants.STATUS_ERROR;
+import static io.coti.basenode.services.BaseNodeServiceManager.networkService;
 import static io.coti.basenode.services.BaseNodeServiceManager.nodeFees;
-import static io.coti.basenode.services.BaseNodeTransactionHelper.CURRENCY_SCALE;
 
 @Slf4j
 @Service
 public class BaseNodeFeesService implements INodeFeesService {
 
-    @Value("${fee.percentage.default:10}")
-    private BigDecimal defaultFeePercentage;
-    @Value("${fee.minimum.default:1}")
-    private BigDecimal defaultMinimumFee;
-    @Value("${fee.maximum.default:25}")
-    private BigDecimal defaultMaximumFee;
-
     @Override
-    public NodeFeeData getNodeFeeData(NodeFeeType nodeFeeType) {
-        Hash nodeFeeHash = CryptoHelper.cryptoHash(nodeFeeType.name().getBytes(StandardCharsets.UTF_8));
+    public TokenFeeData getTokenFeeData(Hash tokenHash, NodeFeeType nodeFeeType) {
+        Hash nodeFeeHash = calculateTokenFeeHash(tokenHash, nodeFeeType);
         return nodeFees.getByHash(nodeFeeHash);
     }
 
+    private void deleteInvalidData() {
+        if (!nodeFees.validate(Object::toString)) {
+            nodeFees.deleteAll();
+            log.warn("Deleted all entries of column family name nodeFees due to invalid data!");
+        }
+    }
+
     @Override
-    public void init(List<NodeFeeType> nodeFeeTypeList) {
-        for (NodeFeeType nodeFeeType : nodeFeeTypeList) {
-            Hash nodeFeeHash = nodeFeeType.getHash();
-            NodeFeeData nodeFeeData = nodeFees.getByHash(nodeFeeHash);
+    public void init(List<TokenFeeData> tokenFeeDataList) {
```

Comment:
there is no validation of tokenFeeData at init, might be not valid values from the properties...